Remove `onPillRemove` prop support from Combo box (#5091)
jpmorganchase/salt-ds: .changeset/stale-dryers-reply.md

---
"@salt-ds/core": patch
---

> Remove `onPillRemove` support from ComboBox API since ComboBox already has an `onSelectionChange` callback which can be used to detect the removed pill.

Reviewer: It's probably worth highlighting that it was only included in documentation only, it technically wasn't removed from the "API"

> Remove `onPillRemove` callback from the ComboBox API. `onPillRemove` callback has never been supported nor is it intended to be supported in the ComboBox as the ComboBox already has an `onSelectionChange` callback which can be used to detect the removed pill.

Reviewer: Changesets should be written in past tense so something like:

```suggestion
Removed `onPillRemove` callback from the ComboBox's type definitions. `onPillRemove` has never been supported nor is it intended to be supported. `onSelectionChange` can be used instead to detect the removed pill.
```
